cardano-cli/cardano-cli.cabal

```diff
@@ -174,28 +174,41 @@ executable cardano-cli
                       , optparse-applicative-fork
                       , transformers-except
 
+library test-lib
+  import:               project-config
+  visibility:           public
+  hs-source-dirs:       test-lib
+  exposed-modules:      Test.OptParse
+  build-depends:        cardano-api
+                      , cardano-cli
+                      , exceptions
+                      , hedgehog
+                      , hedgehog-extras ^>= 0.4.2
+                      , process
+                      , transformers
+
 test-suite cardano-cli-test
-  import:               project-config, maybe-unix, maybe-bytestring
+  import:               project-config, maybe-unix
 
   hs-source-dirs:       test
   main-is:              cardano-cli-test.hs
   type:                 exitcode-stdio-1.0
 
   build-depends:        aeson
                       , base16-bytestring
+                      , bech32 >= 1.1.0
                       , bytestring
-                      , bech32            >= 1.1.0
                       , cardano-api ^>= 8.1.0.1
                       , cardano-api-gen ^>= 8.1.0.2
                       , cardano-cli
+                      , cardano-cli:test-lib
```

Comment:
I think we can build `cabal-version` when we're using newer cabal features.

Reply:
This feature was introduced in `cabal 3.0` and many other projects that this feature are still on `cabal-version: 3.0`.

Reply:
https://cabal.readthedocs.io/en/stable/cabal-package.html#library
>For packages using cabal-version: 3.4 or higher, the syntax to specify an internal library in a build-depends: section is package-name:internal-library-name.